### ISSUE TITLE
Receiving "ERROR: [031]: option 'repo-retention-full-type' must have an index" after converting CentOS8 to Oracle Linux 8

### ISSUE THREAD

**Issue by manderswy** [pgbackrest/pgbackrest#1602]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version:
2.36 (was 2.33 prior to OS switch to Oracle Linux 8)

2. PostgreSQL version:
PostgreSQL 12.9 

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each:
Oracle Linux 8.4

4. Did you install pgBackRest from source or from a package?
package

5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

-pgbackrest.conf.txt attached

-postgresql.conf settings:
archive_command = 'pgbackrest --stanza=carl archive-push %p'
archive_mode = on
listen_addresses = '*'
#max_wal_senders = 10 #not explicitly set
#wal_level = replica #not explicitly set
port = 5432

Error in PostgreSQL log file from incident (recurs every ca. half-second):

ERROR: [031]: option 'repo-retention-full-type' must have an index
2021-12-13 23:59:46.311 MST [1477] LOG:  archive command failed with exit code 31
2021-12-13 23:59:46.311 MST [1477] DETAIL:  The failed archive command was: pgbackrest --stanza=carl archive-push pg_wal/00000003000001DB000000FA
2021-12-13 23:59:46.311 MST [1477] WARNING:  archiving write-ahead log file "00000003000001DB000000FA" failed too many times, will try again later

All .log files in var/log/pgbackrest/ are empty

7. Describe the issue:
pgBackrest was running fine until we used a script provided by Oracle (https://github.com/oracle/centos2ol) to convert a CentOS8 installation to an Oracle Linux 8 installation. After the upgrade, PostgreSQL logs showed the following errors:
[pgbackrest.conf.txt](https://github.com/pgbackrest/pgbackrest/files/7738084/pgbackrest.conf.txt)

ERROR: [031]: option 'repo-retention-full-type' must have an index
2021-12-13 23:59:46.311 MST [1477] LOG:  archive command failed with exit code 31
2021-12-13 23:59:46.311 MST [1477] DETAIL:  The failed archive command was: pgbackrest --stanza=carl archive-push pg_wal/00000003000001DB000000FA
2021-12-13 23:59:46.311 MST [1477] WARNING:  archiving write-ahead log file "00000003000001DB000000FA" failed too many times, will try again later

I am unable to get pgBackrest to run a full backup, expire, or other commands -- all throw the same "ERROR: [031]: option 'repo-retention-full-type' must have an index" error, as above. From the error message, it's not clear to me what the culprit might be.

I recognize that this may be a problem with my installation, but I'm unclear what the exact issue might be, given the error message, and the lack of anything in the logs.

To be clear/come clean, pgBackrest was set up by a previous DBA who departed quickly. I have very limited knowledge of pgBackrest, and I'm scrambling to try to learn more about how it's _supposed_ to work, so I can diagnose.

Backups and archives have my hard drive nearly full, so at a minimum I'm trying to understand how I can purge those so that I have space to breathe and can reinstall and reconfigure pgBackrest, if needed. There is an /archive directory that sits at the same level as our /backup directory, and this directory is 73 GB in size. The /backup folder is 149 GB.

My assumption is that as soon as I can get a backup to run, the WAL files in /archive will be cleared, freeing up space, and the old incremental backup files will also be dropped, which should provide plenty of headspace on disk.

**Comment by jreidthompson** [comment]
Should repo-retention-full-type=count
Be
repo1-retention-full-type=count

**Comment by manderswy** [comment]
> 

Yes, I had actually just made that change, thinking it was a pgBackrest version thing. Interesting, as that setting wasn't causing issues prior to converting to Oracle Linux 8, but maybe things work differently in pgBackrest 2.36 vs. 2.33..? That gets me farther. Now I'm receiving a lock file issue:

ERROR: [050]: unable to acquire lock on file '/tmp/pgbackrest/carl-backup.lock': Permission denied
       HINT: does the user running pgBackRest have permissions on the '/tmp/pgbackrest/carl-backup.lock' file?

Backups have previously been run as the postgres user, and it looks like /tmp/pgbackrest/ is owned by root.

**Comment by jreidthompson** [comment]
To get you further along can you remove the root owned tmp dir and allow backrest to recreate it?  If you can't, you can use the Lock Path Option (--lock-path) option to choose a different location (see the docs for usage)

**Comment by manderswy** [comment]
Thank you! I chowned the /tmp/pgbackrest directory with postgres:postgres. A full backup appears to be running, now. I'm not clear on how long this will take, or if I'm likely to run out of disk space before it completes (presumably, it creates the next full backup before expiring the oldest one, and associated incrementals).

**Comment by manderswy** [comment]
If I've already started the full backup, would it be safe to interrupt it, and instead run an expire on the oldest backup, first, so I was more certain of having sufficient space?

**Comment by jreidthompson** [comment]
Per the docs, I believe that you can use the stop command with the force option to stop an in process backup.  You would then need to issue a start command to remove the stop file before attempting any other action.

**Comment by manderswy** [comment]
That (stop/start) looks to have done the trick. I expired the oldest full, which gave me 126 GB to play with, as expected. Thank you!

**Comment by jreidthompson** [comment]
Great. Glad things appear to be under control.

**Comment by manderswy** [comment]
Greatly appreciate your help! I'll close this out, now. Wondering, though, whether it's documented anywhere that you (apparently) can't mix repo-retention-full-type and repo1-retention-full? I am admittedly new to pgBackrest, but I didn't see documentation on the two versions of each of these settings.

**Comment by dwsteele** [comment]
I checked 2.33 and `repo-retention-full-type` was not valid in that version, though the error message was different:
```
% ./pgbackrest backup --repo-retention-full-type=count
ERROR: [031]: invalid option '--repo-retention-full-type=count'
```
All repo* options introduced after 2.00 require an index. We improved the error message in 2.36 and that's the error you are seeing.

Not sure why you were not seeing it on your old system, but as far as I can see this is not a regression.
